scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1393,344 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def _chandrupatla(func, a, b, *, args=(), xatol=_xtol, xrtol=_rtol,
+                  fatol=None, frtol=0, maxiter=_iter, callback=None):
+    """Find the root of an elementwise function using Chandrupatla's algorithm.
+
+    For each element of the output of `f`, `chandrupatla` seeks the scalar root
+    that makes the element 0. This function allows for `a`, `b`, and the output
+    of `f` to be of any broadcastable shapes.
+
+    Parameters
+    ----------
+
+    func : callable
+        The elementwise function whose root is desired. The signature must be::
+
+            func(x: ndarray, *args) -> ndarray
+
+         where each element of ``x`` is a finite real and ``args`` is a tuple,
+         which may contain an arbitrary number of components of any type(s).
+         `_chandrupatla` seeks an array (or scalar) ``x`` such that each
+         corresponding element in the output array (or scalar) is zero.
+    a, b : array_like
+        The lower and upper bounds of the root of the function. Must be
+        broadcastable with one another.
+    args : tuple, optional
+        Additional positional arguments to be passed to `func`.
```

Comment:
Typically I'd agree, but with these vectorized algorithms there is an important use case for these that can't be taken care of with the other options. I'll explain in real time.

Reply:
Thanks again for discussing this. As we discussed, it will be important for us to have control over `args` and `kwargs` so we can eliminate the elements corresponding with the elements of `x` / `f(x)` that have already converged. As we discussed, that is something planned for a follow-up. It is very easy to do - I just have to add a branch that does *not* re-inflate `x` before evaluating `f(x)` - but we need to discuss the interface for a user to opt into this.